specification/sections/05-events.md

```diff
@@ -175,10 +185,10 @@ see: [static-context paradigm](../glossary.md#static-context-paradigm)
 
 ##### Conditional Requirement 5.3.4.1
 
-> When the provider's `on context changed` is called, the provider **MAY** emit the `PROVIDER_STALE` event, and transition to the `STALE` state.
+> While the provider's `on context changed` function is executing, associated `PROVIDER_CONTEXT_PENDING` handlers **MUST** run.
 
-Some providers cache evaluated flags, and re-evaluate them when the context is changed.
-In these cases, the provider may signal its cache is invalid with the `PROVIDER_STALE` event and the `STALE` provider state.
+The implementation must run any `PROVIDER_CONTEXT_PENDING` handlers associated with the provider while the provider is reconciling its state.
+In languages with asynchronous semantics, the emission of this event can be skipped if the `on context changed` function of the provider in question executes synchronously for a given provider, no other operations can take place while it runs.
```

Comment:
Why are events skipped in `on context changed` is synchronous?

Reply:
If it's synchronous, there's no point emitting a pending event. This would be for providers that don't need to do any async context reconciliation, they just do some local computation. In languages like JS, while that's happening, nothing else can happen, so there' no point in emitting the event. Put another way, the context was reconciled in "real time", so it was never "pending" to the outside world - we'd just skip right to "changed".

In JS, this would look like the `on context change` function of the provider didn't return a promise but just did some synchronous work (returned void). Events would be useless or even potentially cause a disruptive UI flash.

I just wanted to provide this escape hatch here. I'm hesitant to go into more detail in the spec, but I wanted to give implementations the ability to skip this emission because I think it makes sense.